pkg/crc/machine/vsock.go

```diff
@@ -0,0 +1,65 @@
+package machine
+
+import (
+	"fmt"
+
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/daemonclient"
+	"github.com/code-ready/gvisor-tap-vsock/pkg/types"
+	"github.com/pkg/errors"
+)
+
+func exposePorts() error {
+	daemonClient := daemonclient.New()
+	portsToExpose := vsockPorts()
+	alreadyOpenedPorts, err := daemonClient.List()
+	if err != nil {
+		return err
+	}
+	var missingPorts []types.ExposeRequest
+	for _, port := range portsToExpose {
+		if !isOpened(alreadyOpenedPorts, port) {
+			missingPorts = append(missingPorts, port)
+		}
+	}
+	for i := range missingPorts {
+		port := &missingPorts[i]
+		if err := daemonClient.Expose(port); err != nil {
+			return errors.Wrapf(err, "failed to expose port %s -> %s", port.Local, port.Remote)
+		}
```

Comment:
Not sure failure to expose port 443 should be fatal, but this can be addressed later.

Reply:
It happens very early in the process. Even before the VM is started, so it's easy to fix it and launch crc start again. 

If it doesn't work, the console or API wont be reachable, so it's mandatory.

Reply:
Being able to reach the console is not mandatory to use the cluster, is it? People need to be aware the console is broken, but maybe their usecase does not involve the console? For example https://code-ready.github.io/crc/#deploying-sample-application-with-odo_gsg

Reply:
Oh yes, I see! Fair point, we can fail for apiserver and display a (very big) warning for the console port.